src/terms/kinetic.jl

```diff
@@ -40,3 +44,44 @@ end
 
     (E=E, ops=ops)
 end
+
+"""
+Blow-up function as proposed in [REF paper Cancès, Hassan, Vidal to be submitted]
+"""
+function Blowup_function_CHV()
+    g1(x) = sum(abs2, x)
+    # C^2 interpolation part
+    C = [7449.468883604184, -50230.90030429237, 135207.0860245428,
+         -181568.08750464107, 121624.06499834878, -32503.74733526006]
+    g2(x) = C'*((x * ones(6)) .^(0:5))
+    # Blow-up part
+    Ca(a) = (3/2)*(a^2)*(1-a)^(2)
+    a_opti = 0.1792973510040141
+    g3(x) = Ca(a_opti)/( (1-x)^(2) )
+
+    # Assemble all parts
+    function Blowup_function(x)
+        x1, x2 = [0.7, 0.8]
+        (0 ≤ x < x1)   && (return g1(x))
+        (x1 ≤ x < x2)  && (return g2(x))
+        (x2 ≤ x < 1)   && (return g3(x))
+        (x==1) && (return 1e6) # Handle |G+k|^2 = E_cut case
+        error("The blow-up function is defined on [0,1). Did you devide by √Ecut ?")
+    end
+    Blowup_function
+end
+
+"""
+Blow-up function as implemented in the Abinit code [ref code].
+"""
+function Blowup_function_Abinit(Ecut, Ecutsm)
```

Comment:
Ah, now I get it, because you need to pass some extra information.

Ok in that case I would have the blowup stuff be a `struct` (i.e. one standard `BlowupKineticEnergy`, `BlowupCHV` and `BlowupAbinit`). The `BlowupAbinit` one has an `Ecutsm` field, the others have no fields. Then you multiple dispatch on the blowup as in have three methods `TermKinetic(basis::PlaneWaveBasis{T}, blowup::BlowupAbinit)` etc. There you have all the info to build the `TermKinetic` appropriately.